packages/runtime/test/integration/app-update.ts

```diff
@@ -0,0 +1,119 @@
+1. Create 2 transactions in a group: `app_update(n=2), app_update(n=5)`.
+   Check the expected `app.counter == 2, app.total=7`
+   (we need to verify if this is what the Algorand node will do, maybe it will panic?).
+2. Create 2 transactions in a group: `app_update(n=5), app_update(n=2)`.
+   Check the expected `app.counter == 2, app.total=13`
+3. Run tx group: `app_update(n=2), app_update(n=5)` in a loop 1000 times.
+   The expected state should be: `app.counter == 2000`, `app.total = 10 + 500*2 - 500*5`
+ */
+describe("App Update Test", function () {
+  useFixture("app-update");
+  const minBalance = ALGORAND_ACCOUNT_MIN_BALANCE * 10 + 1000; // 1000 to cover fee
+  const john = new AccountStore(1e30);
+  const alice = new AccountStore(minBalance + 1000);
+
+  let runtime: Runtime;
+  let approvalProgram: string;
+  let clearProgram: string;
+  let appId: number;
+  let groupTx: UpdateSSCParam[];
+
+  this.beforeEach(async function () {
+    runtime = new Runtime([john, alice]); // setup test
+    approvalProgram = getProgram('approval_program.teal');
+    clearProgram = getProgram('clear_program.teal');
+    const flags = {
+      sender: john.account,
+      globalBytes: 5,
+      globalInts: 5,
+      localBytes: 5,
+      localInts: 5
+    };
+
+    appId = runtime.addApp(flags, {}, approvalProgram, clearProgram);
+
+    groupTx = [
+      {
+        type: TransactionType.UpdateSSC,
+        sign: SignType.SecretKey,
+        fromAccount: john.account,
+        appID: appId,
+        newApprovalProgram: approvalProgram,
+        newClearProgram: clearProgram,
+        payFlags: {},
+        appArgs: ['int:2']
+      },
+      {
+        type: TransactionType.UpdateSSC,
+        sign: SignType.SecretKey,
+        fromAccount: john.account,
+        appID: appId,
+        newApprovalProgram: approvalProgram,
+        newClearProgram: clearProgram,
+        payFlags: {},
+        appArgs: ['int:5']
+      }
+    ];
+  });
+
+  it("First case", function () {
+    runtime.executeTx(groupTx);
+
+    const globalCounter = runtime.getGlobalState(appId, "counter");
+    const total = runtime.getGlobalState(appId, "total");
+    assert(globalCounter === 2n, "failed counter");
+    assert(total === 7n, "failed total");
+  });
+
+  it("Second case", function () {
+    groupTx[0].appArgs = ['int:5'];
+    groupTx[1].appArgs = ['int:2'];
+
+    runtime.executeTx(groupTx);
+
+    const globalCounter = runtime.getGlobalState(appId, "counter");
+    const total = runtime.getGlobalState(appId, "total");
+    assert(globalCounter === 2n, "failed counter");
+    assert(total === 13n, "failed total");
+  });
+
+  it("Third case", function () {
+    groupTx[0].appArgs = ['int:2'];
+    groupTx[1].appArgs = ['int:5'];
+
+    expectRuntimeError(
+      function () {
+        for (let i = 0; i < 1000; ++i) {
+          runtime.executeTx(groupTx);
+        }
+      },
+      RUNTIME_ERRORS.TEAL.UINT64_UNDERFLOW
+    );
```

Comment:
Please add a comment here (or update test description), that why we get `UINT64_UNDERFLOW`

Reply:
ok